### ISSUE TITLE
native-image failed to build class to executable file.

### ISSUE THREAD

**Issue by junjieyuan** [oracle/graal#4527]
**Describe the issue**
native-image can't build class but can build jar to executable file.

**Steps to reproduce the issue**
Please include both build steps as well as run steps
1. **Write a simple "hello, world" class named HelloWorld.java.**
2. **javac HelloWorld.java**
3. **native-image HelloWorld**

**Describe GraalVM and your environment:**
 - GraalVM version (latest snapshot builds can be found [here](https://github.com/graalvm/graalvm-ce-dev-builds/releases)), or commit id if built from source: **CE 22.1.0**
 - JDK major version: **17**
 - OS: **Fedora Linux Silverblue 36**
 - Architecture: **AMD64**

**More details**
Java source code:
```java
public class HelloWorld {
	public static void main(String... args) {
		System.out.println("hello, world");
	}
}
```

Jar manifest file (HelloWorld.mf):
```text
Manifest-Version: 1.0
Main-Class: HelloWorld
```

Build from class:
```text
[junjie@xps ~]$ native-image --native-image-info --verbose HelloWorld
Executing [
/var/home/junjie/Sdks/graalvm/bin/java \
-XX:+UseParallelGC \
-XX:+UnlockExperimentalVMOptions \
-XX:+EnableJVMCI \
-Dtruffle.TrustAllTruffleRuntimeProviders=true \
-Dtruffle.TruffleRuntime=com.oracle.truffle.api.impl.DefaultTruffleRuntime \
-Dgraalvm.ForcePolyglotInvalid=true \
-Dgraalvm.locatorDisabled=true \
-Dsubstratevm.IgnoreGraalVersionCheck=true \
--add-exports=java.base/com.sun.crypto.provider=ALL-UNNAMED \
--add-exports=java.base/jdk.internal.access.foreign=ALL-UNNAMED \
--add-exports=java.base/jdk.internal.event=ALL-UNNAMED \
--add-exports=java.base/jdk.internal.loader=ALL-UNNAMED \
--add-exports=java.base/jdk.internal.logger=ALL-UNNAMED \
--add-exports=java.base/jdk.internal.misc=ALL-UNNAMED \
--add-exports=java.base/jdk.internal.module=ALL-UNNAMED \
--add-exports=java.base/jdk.internal.org.objectweb.asm=ALL-UNNAMED \
--add-exports=java.base/jdk.internal.perf=ALL-UNNAMED \
--add-exports=java.base/jdk.internal.platform=ALL-UNNAMED \
--add-exports=java.base/jdk.internal.ref=ALL-UNNAMED \
--add-exports=java.base/jdk.internal.reflect=ALL-UNNAMED \
--add-exports=java.base/jdk.internal.vm.annotation=ALL-UNNAMED \
--add-exports=java.base/sun.invoke.util=ALL-UNNAMED \
--add-exports=java.base/sun.net=ALL-UNNAMED \
--add-exports=java.base/sun.nio.ch=ALL-UNNAMED \
--add-exports=java.base/sun.reflect.annotation=ALL-UNNAMED \
--add-exports=java.base/sun.reflect.generics.factory=ALL-UNNAMED \
--add-exports=java.base/sun.reflect.generics.reflectiveObjects=ALL-UNNAMED \
--add-exports=java.base/sun.reflect.generics.repository=ALL-UNNAMED \
--add-exports=java.base/sun.reflect.generics.scope=ALL-UNNAMED \
--add-exports=java.base/sun.reflect.generics.tree=ALL-UNNAMED \
--add-exports=java.base/sun.security.jca=ALL-UNNAMED \
--add-exports=java.base/sun.security.provider=ALL-UNNAMED \
--add-exports=java.base/sun.security.ssl=ALL-UNNAMED \
--add-exports=java.base/sun.security.util=ALL-UNNAMED \
--add-exports=java.base/sun.security.x509=ALL-UNNAMED \
--add-exports=java.base/sun.text.spi=ALL-UNNAMED \
--add-exports=java.base/sun.util.calendar=ALL-UNNAMED \
--add-exports=java.base/sun.util.cldr=ALL-UNNAMED \
--add-exports=java.base/sun.util.locale.provider=ALL-UNNAMED \
--add-exports=java.base/sun.util.locale=ALL-UNNAMED \
--add-exports=java.base/sun.util.resources=ALL-UNNAMED \
--add-exports=java.base/sun.util=ALL-UNNAMED \
--add-exports=java.desktop/sun.java2d.pipe=ALL-UNNAMED \
--add-exports=java.desktop/sun.java2d=ALL-UNNAMED \
--add-exports=java.management/com.sun.jmx.mbeanserver=ALL-UNNAMED \
--add-exports=java.management/sun.management=ALL-UNNAMED \
--add-exports=java.xml.crypto/org.jcp.xml.dsig.internal.dom=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.aarch64=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.amd64=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.code.site=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.code.stack=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.code=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.common=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.hotspot.aarch64=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.hotspot.amd64=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.hotspot=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.meta=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.runtime=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.services=ALL-UNNAMED \
--add-exports=jdk.jfr/jdk.jfr.events=ALL-UNNAMED \
--add-exports=jdk.jfr/jdk.jfr.internal.handlers=ALL-UNNAMED \
--add-exports=jdk.jfr/jdk.jfr.internal.jfc=ALL-UNNAMED \
--add-exports=jdk.jfr/jdk.jfr.internal=ALL-UNNAMED \
--add-exports=jdk.management/com.sun.management.internal=ALL-UNNAMED \
-XX:+UseJVMCINativeLibrary \
-Xss10m \
-Xms1g \
-Xmx6326003296 \
-Djava.awt.headless=true \
-Dorg.graalvm.version=22.1.0 \
-Dorg.graalvm.config=CE \
-Dcom.oracle.graalvm.isaot=true \
-Djava.system.class.loader=com.oracle.svm.hosted.NativeImageSystemClassLoader \
-Xshare:off \
-Djdk.internal.lambda.disableEagerInitialization=true \
-Djdk.internal.lambda.eagerlyInitialize=false \
-Djava.lang.invoke.InnerClassLambdaMetafactory.initializeLambdas=false \
-javaagent:/var/home/junjie/Sdks/graalvm/lib/svm/builder/svm.jar \
-cp \
/var/home/junjie/Sdks/graalvm/lib/svm/builder/svm.jar:/var/home/junjie/Sdks/graalvm/lib/svm/builder/objectfile.jar:/var/home/junjie/Sdks/graalvm/lib/svm/builder/pointsto.jar:/var/home/junjie/Sdks/graalvm/lib/svm/builder/native-image-base.jar:/var/home/junjie/Sdks/graalvm/lib/svm/builder/svm-llvm.jar:/var/home/junjie/Sdks/graalvm/lib/svm/builder/llvm-wrapper-shadowed.jar:/var/home/junjie/Sdks/graalvm/lib/svm/builder/javacpp-shadowed.jar:/var/home/junjie/Sdks/graalvm/lib/svm/builder/llvm-platform-specific-shadowed.jar \
--module-path \
/var/home/junjie/Sdks/graalvm/lib/truffle/truffle-api.jar \
'com.oracle.svm.hosted.NativeImageGeneratorRunner$JDK9Plus' \
-watchpid \
24201 \
-imagecp \
/var/home/junjie:/var/home/junjie/Sdks/graalvm/lib/svm/library-support.jar \
-H:Path=/var/home/junjie \
-H:+DumpTargetInfo \
-H:CLibraryPath=/var/home/junjie/Sdks/graalvm/lib/svm/clibraries/linux-amd64 \
'-H:Class@explicit main-class=HelloWorld' \
'-H:Name@main-class lower case as image name=helloworld'
]
========================================================================================================================
GraalVM Native Image: Generating 'helloworld' (executable)...
========================================================================================================================
[1/7] Initializing...                                                                                   (52.1s @ 0.17GB)
 Version info: 'GraalVM 22.1.0 Java 17 CE'
 C compiler: gcc (redhat, x86_64, 12.0.1)
 Garbage collector: Serial GC
# Printing compilation-target information to: /var/home/junjie/reports/target_info_20220502_233031.txt
[2/7] Performing analysis...  [*]                                                                       (29.8s @ 0.79GB)
   2,798 (73.55%) of  3,804 classes reachable
   3,309 (50.86%) of  6,506 fields reachable
  12,409 (43.60%) of 28,464 methods reachable
      27 classes,     0 fields, and   338 methods registered for reflection

Error: Unable to handle classpath element '/var/home/junjie'. Make sure that all classpath entries are either directories or valid jar files.
Error: Use -H:+ReportExceptionStackTraces to print stacktrace of underlying exception
------------------------------------------------------------------------------------------------------------------------
                        3.2s (3.9% of total time) in 31 GCs | Peak RSS: 1.34GB | CPU load: 1.52
========================================================================================================================
Failed generating 'helloworld' after 1m 22s.
Error: Image build request failed with exit status 1
com.oracle.svm.driver.NativeImage$NativeImageError: Image build request failed with exit status 1
	at com.oracle.svm.driver.NativeImage.showError(NativeImage.java:1678)
	at com.oracle.svm.driver.NativeImage.build(NativeImage.java:1389)
	at com.oracle.svm.driver.NativeImage.performBuild(NativeImage.java:1350)
	at com.oracle.svm.driver.NativeImage.main(NativeImage.java:1337)
```

Build from jar:
```text
[junjie@xps ~]$ native-image --native-image-info --verbose -jar Hello.jar 
Executing [
/var/home/junjie/Sdks/graalvm/bin/java \
-XX:+UseParallelGC \
-XX:+UnlockExperimentalVMOptions \
-XX:+EnableJVMCI \
-Dtruffle.TrustAllTruffleRuntimeProviders=true \
-Dtruffle.TruffleRuntime=com.oracle.truffle.api.impl.DefaultTruffleRuntime \
-Dgraalvm.ForcePolyglotInvalid=true \
-Dgraalvm.locatorDisabled=true \
-Dsubstratevm.IgnoreGraalVersionCheck=true \
--add-exports=java.base/com.sun.crypto.provider=ALL-UNNAMED \
--add-exports=java.base/jdk.internal.access.foreign=ALL-UNNAMED \
--add-exports=java.base/jdk.internal.event=ALL-UNNAMED \
--add-exports=java.base/jdk.internal.loader=ALL-UNNAMED \
--add-exports=java.base/jdk.internal.logger=ALL-UNNAMED \
--add-exports=java.base/jdk.internal.misc=ALL-UNNAMED \
--add-exports=java.base/jdk.internal.module=ALL-UNNAMED \
--add-exports=java.base/jdk.internal.org.objectweb.asm=ALL-UNNAMED \
--add-exports=java.base/jdk.internal.perf=ALL-UNNAMED \
--add-exports=java.base/jdk.internal.platform=ALL-UNNAMED \
--add-exports=java.base/jdk.internal.ref=ALL-UNNAMED \
--add-exports=java.base/jdk.internal.reflect=ALL-UNNAMED \
--add-exports=java.base/jdk.internal.vm.annotation=ALL-UNNAMED \
--add-exports=java.base/sun.invoke.util=ALL-UNNAMED \
--add-exports=java.base/sun.net=ALL-UNNAMED \
--add-exports=java.base/sun.nio.ch=ALL-UNNAMED \
--add-exports=java.base/sun.reflect.annotation=ALL-UNNAMED \
--add-exports=java.base/sun.reflect.generics.factory=ALL-UNNAMED \
--add-exports=java.base/sun.reflect.generics.reflectiveObjects=ALL-UNNAMED \
--add-exports=java.base/sun.reflect.generics.repository=ALL-UNNAMED \
--add-exports=java.base/sun.reflect.generics.scope=ALL-UNNAMED \
--add-exports=java.base/sun.reflect.generics.tree=ALL-UNNAMED \
--add-exports=java.base/sun.security.jca=ALL-UNNAMED \
--add-exports=java.base/sun.security.provider=ALL-UNNAMED \
--add-exports=java.base/sun.security.ssl=ALL-UNNAMED \
--add-exports=java.base/sun.security.util=ALL-UNNAMED \
--add-exports=java.base/sun.security.x509=ALL-UNNAMED \
--add-exports=java.base/sun.text.spi=ALL-UNNAMED \
--add-exports=java.base/sun.util.calendar=ALL-UNNAMED \
--add-exports=java.base/sun.util.cldr=ALL-UNNAMED \
--add-exports=java.base/sun.util.locale.provider=ALL-UNNAMED \
--add-exports=java.base/sun.util.locale=ALL-UNNAMED \
--add-exports=java.base/sun.util.resources=ALL-UNNAMED \
--add-exports=java.base/sun.util=ALL-UNNAMED \
--add-exports=java.desktop/sun.java2d.pipe=ALL-UNNAMED \
--add-exports=java.desktop/sun.java2d=ALL-UNNAMED \
--add-exports=java.management/com.sun.jmx.mbeanserver=ALL-UNNAMED \
--add-exports=java.management/sun.management=ALL-UNNAMED \
--add-exports=java.xml.crypto/org.jcp.xml.dsig.internal.dom=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.aarch64=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.amd64=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.code.site=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.code.stack=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.code=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.common=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.hotspot.aarch64=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.hotspot.amd64=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.hotspot=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.meta=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.runtime=ALL-UNNAMED \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.services=ALL-UNNAMED \
--add-exports=jdk.jfr/jdk.jfr.events=ALL-UNNAMED \
--add-exports=jdk.jfr/jdk.jfr.internal.handlers=ALL-UNNAMED \
--add-exports=jdk.jfr/jdk.jfr.internal.jfc=ALL-UNNAMED \
--add-exports=jdk.jfr/jdk.jfr.internal=ALL-UNNAMED \
--add-exports=jdk.management/com.sun.management.internal=ALL-UNNAMED \
-XX:+UseJVMCINativeLibrary \
-Xss10m \
-Xms1g \
-Xmx6326003296 \
-Djava.awt.headless=true \
-Dorg.graalvm.version=22.1.0 \
-Dorg.graalvm.config=CE \
-Dcom.oracle.graalvm.isaot=true \
-Djava.system.class.loader=com.oracle.svm.hosted.NativeImageSystemClassLoader \
-Xshare:off \
-Djdk.internal.lambda.disableEagerInitialization=true \
-Djdk.internal.lambda.eagerlyInitialize=false \
-Djava.lang.invoke.InnerClassLambdaMetafactory.initializeLambdas=false \
-javaagent:/var/home/junjie/Sdks/graalvm/lib/svm/builder/svm.jar \
-cp \
/var/home/junjie/Sdks/graalvm/lib/svm/builder/svm.jar:/var/home/junjie/Sdks/graalvm/lib/svm/builder/objectfile.jar:/var/home/junjie/Sdks/graalvm/lib/svm/builder/pointsto.jar:/var/home/junjie/Sdks/graalvm/lib/svm/builder/native-image-base.jar:/var/home/junjie/Sdks/graalvm/lib/svm/builder/svm-llvm.jar:/var/home/junjie/Sdks/graalvm/lib/svm/builder/llvm-wrapper-shadowed.jar:/var/home/junjie/Sdks/graalvm/lib/svm/builder/javacpp-shadowed.jar:/var/home/junjie/Sdks/graalvm/lib/svm/builder/llvm-platform-specific-shadowed.jar \
--module-path \
/var/home/junjie/Sdks/graalvm/lib/truffle/truffle-api.jar \
'com.oracle.svm.hosted.NativeImageGeneratorRunner$JDK9Plus' \
-watchpid \
27580 \
-imagecp \
/var/home/junjie/Hello.jar:/var/home/junjie/Sdks/graalvm/lib/svm/library-support.jar \
-H:Path=/var/home/junjie \
-H:+DumpTargetInfo \
'-H:Class@manifest from file:///var/home/junjie/Hello.jar=HelloWorld' \
'-H:Name@manifest from file:///var/home/junjie/Hello.jar=Hello' \
-H:CLibraryPath=/var/home/junjie/Sdks/graalvm/lib/svm/clibraries/linux-amd64
]
========================================================================================================================
GraalVM Native Image: Generating 'Hello' (executable)...
========================================================================================================================
[1/7] Initializing...                                                                                    (6.0s @ 0.18GB)
 Version info: 'GraalVM 22.1.0 Java 17 CE'
 C compiler: gcc (redhat, x86_64, 12.0.1)
 Garbage collector: Serial GC
# Printing compilation-target information to: /var/home/junjie/reports/target_info_20220502_233419.txt
# Printing native-library information to: /var/home/junjie/reports/native_library_info_20220502_233440.txt
[2/7] Performing analysis...  [******]                                                                  (20.6s @ 1.16GB)
   2,852 (74.17%) of  3,845 classes reachable
   3,399 (50.83%) of  6,687 fields reachable
  12,932 (44.56%) of 29,019 methods reachable
      27 classes,     0 fields, and   345 methods registered for reflection
      57 classes,    58 fields, and    51 methods registered for JNI access
[3/7] Building universe...                                                                               (3.6s @ 0.47GB)
[4/7] Parsing methods...      [*]                                                                        (1.3s @ 0.72GB)
[5/7] Inlining methods...     [****]                                                                     (1.8s @ 1.44GB)
[6/7] Compiling methods...    [[6/7] Compiling methods...    [****]                                                                    (16.6s @ 1.71GB)
[7/7] Creating image...                                                                                  (2.0s @ 1.98GB)
   4.41MB (35.35%) for code area:    7,591 compilation units
   6.97MB (55.89%) for image heap:   1,711 classes and 93,028 objects
   1.09MB ( 8.76%) for other data
  12.47MB in total
------------------------------------------------------------------------------------------------------------------------
Top 10 packages in code area:                               Top 10 object types in image heap:
 666.76KB java.util                                         1001.32KB byte[] for general heap data
 338.22KB java.lang                                          944.62KB byte[] for code metadata
 274.64KB java.text                                          877.88KB java.lang.String
 235.70KB java.util.regex                                    623.30KB java.lang.Class
 194.32KB com.oracle.svm.jni                                 532.49KB byte[] for java.lang.String
 193.32KB java.util.concurrent                               388.08KB java.util.HashMap$Node
 157.08KB com.oracle.svm.core.reflect                        222.81KB com.oracle.svm.core.hub.DynamicHubCompanion
 147.35KB java.math                                          167.69KB java.util.HashMap$Node[]
  98.45KB java.util.logging                                  163.43KB java.lang.String[]
  94.89KB com.oracle.svm.core.genscavenge                    156.14KB java.util.concurrent.ConcurrentHashMap$Node
      ... 119 additional packages                                 ... 776 additional object types
                                           (use GraalVM Dashboard to see all)
------------------------------------------------------------------------------------------------------------------------
                        2.8s (5.0% of total time) in 18 GCs | Peak RSS: 2.57GB | CPU load: 3.52
------------------------------------------------------------------------------------------------------------------------
Produced artifacts:
 /var/home/junjie/Hello (executable)
 /var/home/junjie/Hello.build_artifacts.txt
========================================================================================================================
Finished generating 'Hello' in 53.9s.
```

**Comment by oubidar-Abderrahim** [comment]
Hi, Thank you for reporting this, we will take a look into it and get back to you

**Comment by oubidar-Abderrahim** [comment]
Hi @junjieyuan, I tested locally and was able to build the hello world example using GraalVM CE java17 22.1.0 on my Linux amd64 machine.
I think there's an issue with the installation path on your machine, please check the permissions on your folder, and maybe use a different parent folder for your test case.

**Comment by junjieyuan** [comment]
> Hi @junjieyuan, I tested locally and was able to build the hello world example using GraalVM CE java17 22.1.0 on my Linux amd64 machine. I think there's an issue with the installation path on your machine, please check the permissions on your folder, and maybe use a different parent folder for your test case.

Hello, @oubidar-Abderrahim. Thank you for you help.

**TLDR:**

**I tried creating a new directory and in the new directory it compiles successfully.**

I initially confirmed what caused the compilation to fail:
It may be that the permission restrictions on the files in my podman storage volume directory ($HOME/.local/share/containers/storage/volumes) are causing the IOException to be thrown.

Actually /var/home/junjie is my home directory, the system I use is Fedora Silverblue, and /home/junjie is its soft link. There are many files in the home directory, and native-image will try to traverse them and add classes to the classpath.

https://github.com/oracle/graal/blob/f07c03ff6f04c303addbcd8226d6f82f48e7e98e/substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/ClassLoaderSupportImpl.java#L125-L176

Line 147
https://github.com/oracle/graal/blob/f07c03ff6f04c303addbcd8226d6f82f48e7e98e/substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/ClassLoaderSupportImpl.java#L147

and 156
https://github.com/oracle/graal/blob/f07c03ff6f04c303addbcd8226d6f82f48e7e98e/substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/ClassLoaderSupportImpl.java#L156

may throw IOException. So I simpified the logic to verify:

```java
import java.io.IOException;
import java.io.InputStream;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.ArrayDeque;
import java.util.stream.Stream;

public class Test {
    public static void main(String[] args) throws IOException {
        // var root = Path.of("/var/home/junjie/.local/share/containers");
        var root = Path.of("/var/home/junjie/test");
        ArrayDeque<Path> queue = new ArrayDeque<>();
        queue.push(root);
        while (!queue.isEmpty()) {
            Path entry = queue.pop();

            if (Files.isDirectory(entry)) {
                try (Stream<Path> pathStream = Files.list(entry)) {
                    pathStream.forEach(queue::push);
                }
            } else {
                try (InputStream is = Files.newInputStream(entry)) {
                } catch (IOException e) {
			e.printStackTrace();
		}
            }
        }
    }
}
```

Verify result (in my home directory) :
* broken soft link (a soft link point to not exist file) will throw IOException
* no UNIX or SELinux permission will throw IOException

There may be more, but I have limited energy to test them all.

But when I create a file or a soft link to verify the conjecture I have a problem:
Broken symlink files I created, files without UNIX or SELinux permissions, and symlinks to unpermissioned files in the container storage volume (eg binlog.000115 in MySql volume) did not reproduce the problem, only to directories outside the storage volume (eg $HOME/.local/share /containers/storage/volumes/mysql-data, my MySql container's volume data in it) can reproduce the problem. It may be because my simplified logic is inconsistent with the original logic.

If anyone is curious about file permissions in storage volumes, take a look at:

```text
⬢[junjie@toolbox ~]$ ls -lZR .local/share/containers/storage/volumes/mysql-data
.local/share/containers/storage/volumes/mysql-data:
total 0
drwxrwxrwt. 1 systemd-oom systemd-oom system_u:object_r:container_file_t:s0 950  4月 13 17:57 _data

.local/share/containers/storage/volumes/mysql-data/_data:
total 181980
-rw-r-----. 1 systemd-oom systemd-oom system_u:object_r:container_file_t:s0       56  9月  1  2021  auto.cnf
-rw-r-----. 1 systemd-oom systemd-oom system_u:object_r:container_file_t:s0      157  3月 14 23:44  binlog.000115
-rw-r-----. 1 systemd-oom systemd-oom system_u:object_r:container_file_t:s0      180  3月 15 22:35  binlog.000116
-rw-r-----. 1 systemd-oom systemd-oom system_u:object_r:container_file_t:s0      180  3月 15 23:47  binlog.000117
-rw-r-----. 1 systemd-oom systemd-oom system_u:object_r:container_file_t:s0      157  3月 17 23:17  binlog.000118
-rw-r-----. 1 systemd-oom systemd-oom system_u:object_r:container_file_t:s0      157  3月 19 00:11  binlog.000119
-rw-r-----. 1 systemd-oom systemd-oom system_u:object_r:container_file_t:s0      157  3月 20 12:51  binlog.000120
-rw-r-----. 1 systemd-oom systemd-oom system_u:object_r:container_file_t:s0      157  3月 21 18:26  binlog.000121
-rw-r-----. 1 systemd-oom systemd-oom system_u:object_r:container_file_t:s0      157  3月 22 13:43  binlog.000122
-rw-r-----. 1 systemd-oom systemd-oom system_u:object_r:container_file_t:s0      157  3月 23 20:48  binlog.000123
-rw-r-----. 1 systemd-oom systemd-oom system_u:object_r:container_file_t:s0      157  3月 26 16:23  binlog.000124
-rw-r-----. 1 systemd-oom systemd-oom system_u:object_r:container_file_t:s0      157  3月 28 20:10  binlog.000125
-rw-r-----. 1 systemd-oom systemd-oom system_u:object_r:container_file_t:s0      157  3月 30 10:46  binlog.000126
-rw-r-----. 1 systemd-oom systemd-oom system_u:object_r:container_file_t:s0      157  4月 13 17:57  binlog.000127
-rw-r-----. 1 systemd-oom systemd-oom system_u:object_r:container_file_t:s0      157  4月 13 17:57  binlog.000128
-rw-r-----. 1 systemd-oom systemd-oom system_u:object_r:container_file_t:s0      224  4月 13 17:57  binlog.index
-rw-------. 1 systemd-oom systemd-oom system_u:object_r:container_file_t:s0     1676  9月  1  2021  ca-key.pem
-rw-r--r--. 1 systemd-oom systemd-oom system_u:object_r:container_file_t:s0     1112  9月  1  2021  ca.pem
-rw-r--r--. 1 systemd-oom systemd-oom system_u:object_r:container_file_t:s0     1112  9月  1  2021  client-cert.pem
-rw-------. 1 systemd-oom systemd-oom system_u:object_r:container_file_t:s0     1680  9月  1  2021  client-key.pem
drwxr-x---. 1 systemd-oom systemd-oom system_u:object_r:container_file_t:s0       76  1月 25 23:18  cms
drwxr-x---. 1 systemd-oom systemd-oom system_u:object_r:container_file_t:s0       22  3月  1 23:36  cun
-rw-r-----. 1 systemd-oom systemd-oom system_u:object_r:container_file_t:s0   196608  4月 13 17:59 '#ib_16384_0.dblwr'
-rw-r-----. 1 systemd-oom systemd-oom system_u:object_r:container_file_t:s0  8585216  1月 21 00:02 '#ib_16384_1.dblwr'
-rw-r-----. 1 systemd-oom systemd-oom system_u:object_r:container_file_t:s0     4200  3月 15 23:47  ib_buffer_pool
-rw-r-----. 1 systemd-oom systemd-oom system_u:object_r:container_file_t:s0 12582912  4月 13 17:57  ibdata1
-rw-r-----. 1 systemd-oom systemd-oom system_u:object_r:container_file_t:s0 50331648  4月 13 17:59  ib_logfile0
-rw-r-----. 1 systemd-oom systemd-oom system_u:object_r:container_file_t:s0 50331648  4月 13 17:59  ib_logfile1
-rw-r-----. 1 systemd-oom systemd-oom system_u:object_r:container_file_t:s0 12582912  4月 13 17:57  ibtmp1
drwxr-x---. 1 systemd-oom systemd-oom system_u:object_r:container_file_t:s0      202  4月 13 17:57 '#innodb_temp'
drwxr-x---. 1 systemd-oom systemd-oom system_u:object_r:container_file_t:s0      178  1月 21 00:02  mysql
-rw-r-----. 1 systemd-oom systemd-oom system_u:object_r:container_file_t:s0 46137344  4月 13 17:57  mysql.ibd
-rw-r-----. 1 systemd-oom systemd-oom system_u:object_r:container_file_t:s0        6  1月 21 00:02  mysql_upgrade_info
drwxr-x---. 1 systemd-oom systemd-oom system_u:object_r:container_file_t:s0     5068  1月 21 00:02  performance_schema
-rw-------. 1 systemd-oom systemd-oom system_u:object_r:container_file_t:s0     1680  9月  1  2021  private_key.pem
-rw-r--r--. 1 systemd-oom systemd-oom system_u:object_r:container_file_t:s0      452  9月  1  2021  public_key.pem
-rw-r--r--. 1 systemd-oom systemd-oom system_u:object_r:container_file_t:s0     1112  9月  1  2021  server-cert.pem
-rw-------. 1 systemd-oom systemd-oom system_u:object_r:container_file_t:s0     1680  9月  1  2021  server-key.pem
drwxr-x---. 1 systemd-oom systemd-oom system_u:object_r:container_file_t:s0       28  9月  1  2021  sys
-rw-r-----. 1 systemd-oom systemd-oom system_u:object_r:container_file_t:s0 16777216  4月 13 17:59  undo_001
-rw-r-----. 1 systemd-oom systemd-oom system_u:object_r:container_file_t:s0 16777216  4月 13 17:59  undo_002
ls: cannot open directory '.local/share/containers/storage/volumes/mysql-data/_data/cms': Permission denied
ls: cannot open directory '.local/share/containers/storage/volumes/mysql-data/_data/cun': Permission denied
ls: cannot open directory '.local/share/containers/storage/volumes/mysql-data/_data/#innodb_temp': Permission denied
ls: cannot open directory '.local/share/containers/storage/volumes/mysql-data/_data/mysql': Permission denied
ls: cannot open directory '.local/share/containers/storage/volumes/mysql-data/_data/performance_schema': Permission denied
ls: cannot open directory '.local/share/containers/storage/volumes/mysql-data/_data/sys': Permission denied
```
